azure/services/managedclusters/spec_test.go

```diff
@@ -485,6 +485,92 @@ func TestParameters(t *testing.T) {
 				g.Expect(*result.(armcontainerservice.ManagedCluster).Properties.OidcIssuerProfile.Enabled).To(BeFalse())
 			},
 		},
+		{
+			name:     "do not update addon profile",
+			existing: getExistingClusterWithAddonProfile(),
+			spec: &ManagedClusterSpec{
+				Name:          "test-managedcluster",
+				ResourceGroup: "test-rg",
+				Location:      "test-location",
+				Tags: map[string]string{
+					"test-tag": "test-value",
+				},
+				Version:         "v1.22.0",
+				LoadBalancerSKU: "standard",
+				AddonProfiles: []AddonProfile{
+					{
+						Name:    "first-addon-profile",
+						Enabled: true,
+					},
+					{
+						Name:    "second-addon-profile",
+						Enabled: true,
+					},
+				},
+			},
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeAssignableToTypeOf(armcontainerservice.ManagedCluster{}))
+				g.Expect(*result.(armcontainerservice.ManagedCluster).Properties.AddonProfiles["first-addon-profile"].Enabled).To(BeTrue())
```

Comment:
If this test is making sure we don't issue an update when addon profiles are equivalent, shouldn't `result` be nil?

Reply:
yes you are right, good catch !!